python/az/aro/azext_aro/_dynamic_validators.py

```diff
@@ -0,0 +1,286 @@
+                "Microsoft.Network/virtualNetworks/write",
+                "Microsoft.Network/virtualNetworks/subnets/join/action",
+                "Microsoft.Network/virtualNetworks/subnets/read",
+                "Microsoft.Network/virtualNetworks/subnets/write", ])
+
+        return errors
+
+    return _validate_vnet
+
+
+def dyn_validate_subnet(key):
+    def _validate_subnet(cmd, namespace):
+        errors = []
+        with AutomaticSpinner(f"Validating {key} permissions"):
+            subnet = getattr(namespace, key)
+
+            if not is_valid_resource_id(subnet):
+                if not namespace.vnet:
+                    raise RequiredArgumentMissingError(
+                        f"Must specify --vnet if --{key.replace('_', '-')} is not an id.")
+
+                validate_vnet(cmd, namespace)
+
+                subnet = namespace.vnet + '/subnets/' + subnet
+                setattr(namespace, key, subnet)
+
+            parts = parse_resource_id(subnet)
+
+            network_client = get_mgmt_service_client(
+                cmd.cli_ctx, ResourceType.MGMT_NETWORK)
+
+            auth_client = get_mgmt_service_client(
+                cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION, api_version="2015-07-01")
+
+            try:
+                subnet_obj = network_client.subnets.get(parts['resource_group'],
+                                                        parts['name'],
+                                                        parts['child_name_1'])
+
+                route_table_obj = subnet_obj.route_table
+                if route_table_obj is None:
+                    raise ResourceNotFoundError("Subnet is missing route table")
+            except Exception as err:
+                if isinstance(err, ResourceNotFoundError):
+                    raise InvalidArgumentValueError(
+                        f"Invalid --{key.replace('_', '-')}, error when getting '{subnet}': {str(err)}") from err
+                raise CLIInternalError(
+                    f"Unexpected error when getting subnet '{subnet}': {str(err)}") from err
+
+            route_parts = parse_resource_id(route_table_obj.id)
+
+            errors = validate_resource(auth_client, f"{key}_route_table", route_parts, [
+                "Microsoft.Network/routeTables/join/action",
+                "Microsoft.Network/routeTables/read",
+                "Microsoft.Network/routeTables/write"])
+
+            if subnet_obj.network_security_group is not None:
+                message = f"A Network Security Group \"{subnet_obj.network_security_group}\" "\
+                          "is already assigned to this subnet. Ensure there a no Network "\
+                          "Security Groups assigned to cluster subnets before cluster creation"
+                error = [f"{key}", parts['child_name_1'], message]
+                errors.append(error)
+
+        return errors
+    return _validate_subnet
+
+
+def dyn_validate_cidr_ranges():
```

Comment:
Would it make sense to also check if the pod cidr is large enough to provision master nodes like we do in basic checks?

Reply:
Added this in!